flink-end-to-end-tests/flink-sql-gateway-test/src/test/java/org/apache/flink/table/gateway/SqlGatewayE2ECase.java

```diff
@@ -91,36 +94,52 @@ public static void beforeClass() {
 
     @AfterClass
     public static void afterClass() throws Exception {
-        port.close();
+        hiveserver2Port.close();
+        restPort.close();
     }
 
     @Test
-    public void testExecuteStatement() throws Exception {
-        URL url = SqlGatewayE2ECase.class.getClassLoader().getResource(GATEWAY_E2E_SQL);
-        if (url == null) {
-            throw new FileNotFoundException(GATEWAY_E2E_SQL);
-        }
+    public void testHiveserver2ExecuteStatement() throws Exception {
         File result = FOLDER.newFolder("csv");
-        String sql =
-                Files.readAllLines(new File(url.getFile()).toPath()).stream()
-                        .filter(line -> !line.trim().startsWith("--"))
-                        .collect(Collectors.joining());
-        List<String> lines =
-                Arrays.stream(sql.split(";"))
-                        .map(line -> line.replace(RESULT_KEY, result.getAbsolutePath()))
-                        .collect(Collectors.toList());
-
         try (GatewayController gateway = flinkResource.startSqlGateway();
                 ClusterController ignore = flinkResource.startCluster(1)) {
             gateway.submitSQLJob(
-                    new SQLJobSubmission.SQLJobSubmissionBuilder(lines)
+                    new SQLJobSubmission.SQLJobSubmissionBuilder(getSqlLines(result))
                             .setClientMode(SQLJobSubmission.ClientMode.HIVE_JDBC)
                             .build(),
                     Duration.ofSeconds(60));
         }
         assertEquals(Collections.singletonList("1"), readCsvResultFiles(result.toPath()));
     }
 
+    @Test
+    public void testRestExecuteStatement() throws Exception {
+        File result = FOLDER.newFolder("csv");
+        try (GatewayController gateway = flinkResource.startSqlGateway();
+                ClusterController ignore = flinkResource.startCluster(1)) {
+            gateway.submitSQLJob(
+                    new SQLJobSubmission.SQLJobSubmissionBuilder(getSqlLines(result))
+                            .setClientMode(SQLJobSubmission.ClientMode.REST)
+                            .build(),
+                    Duration.ofSeconds(60));
+        }
+        assertEquals(Collections.singletonList("1"), readCsvResultFiles(result.toPath()));
```

Comment:
The `testRestExecuteStatement` and `testHiveserver2ExecuteStatement` is much similar. Can we add a helper method like `runExecuteStatement(ClientMode clientMode)`